integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/PanacheMockingTest.java

```diff
@@ -0,0 +1,134 @@
+package io.quarkus.it.panache;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.persistence.LockModeType;
+import javax.ws.rs.WebApplicationException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+import io.quarkus.panache.mock.PanacheMock;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+@QuarkusTest
+public class PanacheMockingTest {
+
+    @Test
+    @Order(1)
+    public void testPanacheMocking() {
+        PanacheMock.mock(Person.class);
+
+        Assertions.assertEquals(0, Person.count());
+
+        Mockito.when(Person.count()).thenReturn(23l);
+        Assertions.assertEquals(23, Person.count());
+
+        Mockito.when(Person.count()).thenReturn(42l);
+        Assertions.assertEquals(42, Person.count());
+
+        Mockito.when(Person.count()).thenCallRealMethod();
+        Assertions.assertEquals(0, Person.count());
+
+        PanacheMock.verify(Person.class, Mockito.times(4)).count();
+
+        Person p = new Person();
+        Mockito.when(Person.findById(12l)).thenReturn(p);
```

Comment:
Can you add some tests with multiple long/double method parameters? It should show up the issue I brought up above. These tests don't show it because there is only a single parameter.

Reply:
I can't, see above.

Reply:
Actually I can make user methods do this. Sorry.